### ISSUE TITLE
Improve support for older geometries

### ISSUE THREAD

**Issue by CanyonFoot** [vengroff/censusdis#158]
While trying to retrieve the 2010 census tract boundaries, I ran into an error. I think there are actually two distinct issues so I'll go over the different cases.

***Incorrect column names***

Running
`gdf_tract = ced.download(
    "acs/acs5",
    2010,
    ["NAME"],
    state="01",
    county = "*",
    tract = "*",
    with_geometry=True
)`

Errors with: 

`KeyError: "['TRACTCE'] not in index"`

The reason is that CB tract files on the TIGER database identify the tracts using a column called 'TRACT', rather than the newer files that use 'TRACTCE'. This issue is not present at the state or county level, but something similar happens with block groups (e.g., old: 'BLKGRP', new: 'BLKGRPCE'). This should be quite easy to fix - I think it's enough just to rename the relevant columns when they appear such that the dataframe always contains the expected version of these columns.

***Inconsistent TIGER folder structure, missing years***

The TIGER FTP server contains several different versions of the geographies in each year. The directory/naming conventions have been consistent for the last ~9 years, but before 2014, there are some differences that make the urls used by censusdis invalid. For instance,

`gdf_tract = ced.download(
    "acs/acs5",  
    2013,
    ["NAME"],
    state="01",
    county = "*",
    tract = "*",
    with_geometry=True
)`

Fails with:

`MapException: Bad zip file retrieved from https://www2.census.gov/geo/tiger/GENZ2013/shp/cb_2013_01_tract_500k.zip`

And similar for 2011, 2012, and any year prior to 2010 (2010 works, other than the column issue, likely because it was a decennial census year). This [PDF](https://www2.census.gov/geo/tiger/Directory_Contents_ReadMe.pdf) doesn't seem to have been updated since 2018, but gives an overview of what's available for previous years. Note that the cartographic boundary folders for some years are missing (i.e. 2011, anything before 2010), but 2012 and 2013 are both present in the directory. If you look inside one of these folders such as [GENZ2013](https://www2.census.gov/geo/tiger/GENZ2013/) you'll see that there is no subdirectory `shp`, which is why we get the error above.

I think there are a couple of approaches to consider for resolving this problem.

For 2012 and 2013, it would be possible to just use the url without the `shp` subdirectory. However, this would not work for 2011 or any of the years before 2010 since no CB folders exist for those years (except in decennial years 2000 and 1990). The full-size (i.e. not CB) files exist going back to the mid-2000s, so it would be possible to read these in for years without a CB file, although they are substantially larger (about 5x) than the CB files. However, there are further directory structure differences in these folders that would require some finicky adjustments to ensure everything works.

A potentially simpler approach would be to load the CB files for the most recent decennial census in any year that doesn't have a normal `GENZ**` folder. So querying 2013 tract data would retrieve the 2010 CB files and query 2008 would retrieve the 2000 CB files. This method relies on the assumption that the boundaries don't change except during the decennial census which is definitely _mostly_ true but there could be some weird edge cases where boundaries change between decennial censuses in which case this approach could be problematic.  I think it's worth investigating this a little more because if that assumption holds it would be a much easier solution than any of the others.

**Comment by vengroff** [comment]
This is a good catch, and I think you are right, there are two issues here. It means the `censusdis` code has to 
get a little hairier and have more special cases, but it's goal is to present a clean interface, so it had to hide some
of this internal inconsistency. If you look in `map.py` you will already see some special case code for dealing with
pre and post 2010 data. It handles the `shp` component of the path being there or not. Looks like we will need a little more of that flavor to handle 2012 and 2013 properly. I think I'd rather do this than just use the decenial shapefiles. 

I do see that the GENZ2011 directory is missing from https://www2.census.gov/geo/tiger/. Maybe we have to fall back on TIGER2011 instead of CB in that one case?

For the other part (the first half) of this issue, I think you are right we just have to catch it in the right place and 
map the name. Of course, it is year dependent. I created a branch, wrote a simple test case, and did a couple of
changes to handle this. See https://github.com/vengroff/censusdis/tree/issue/158/old-geo. 

For pre-2010, I think since there are no shapefiles, that we have two options. Either use the 2010 shapefiles or 
throw and error and tell the user that `with_geometry=True` is only supported for 2010 and onward. Or maybe 
we raise an exception for 2011 also and say geometry is available only for 2010 and 2012 through the present (2022)
as of this writing.

I'd be happy to have you work on the rest of it. You can check out that branch and go from there.

**Comment by CanyonFoot** [comment]
I think that all makes sense. I noticed that there was already some code in `map.py` to rename STATE and COUNTY in the older shapefiles so I added TRACT and BLKGRP to that rename call and removed the code in `data.py` to keep the name stuff in one place. 

My current use case just relies on those 2010 files so if you don't mind adding this change for a new version I'd be happy to post a separate issue going through some of the more complicated cases and work on that over the next couple of weeks.

**Comment by vengroff** [comment]
That sounds good to me. Please create a pull request for the 2010 stuff and we can unblock you by releasing that, then we can decide what to do with the other stuff. It's complicated because some years there just aren't shapefiles. See the year tabs and "More" dropdown in https://www.census.gov/geographies/mapping-files/time-series/geo/carto-boundary-file.html for example. Or the directories of files they point to, as you pointed out earlier.
